docker/Dockerfile

```diff
@@ -1,5 +1,11 @@
 FROM hpcaitech/cuda-conda:11.3
 
+# metainformation
+LABEL org.opencontainers.image.version = "0.2.5"
```

Comment:
Can this be made auto-fetched instead of hard-coded? The version is always specified in the `version.txt`.

Reply:
One possible solution could be to have inline python i.e.

```Dockerfile
LABEL org.opencontainers.image.version = $(python -c "import colossalai; print(colossalai.__version__)")
```

---

Another option could be to pass the version as an argument while building the docker image i.e.

```Dockerfile
ARG PKGVERSION=X.X.X
...
LABEL org.opencontainers.image.version = ${BASE_IMAGE}
```

and then build using

```bash
docker build --build-arg PKGVERSION= <...>
```

Reply:
I personally think the last method is the easiest for the following reaons:
1. For the first method, since `colossalai` is installed  after the `LABEL` declaration, we cannot import `colossalai` when fetching the version ahead of time
2. For the second method, I find it feasible but not really user friendly since many users directly run `docker build` and  do not take care about the arguments much. The second method requires the user to check the content of `version.txt` and then do `docker build`, which might be troublesome for users who are not familiar with this requirement.

As the last method is really automatic, I would recommend to proceed with it. Feel free to share your thoughts as well. 😃

Reply:
On a side note, we are making our development plan public this week. We'd really like to invite you to be part of the development if you are interested.

Reply:
@FrankLeeeee that sounds interesting. How can I get involved ???

Reply:
Also, if we plan to use the last method, I would really like to use the `get_version()` fn from `setup.py` but that's not really inline code. Can we just move the `version` label until after `colossalai` is installed ?

Reply:
Hey, @SauravMaheshkar , you just join our development channel via the link on this page.

https://github.com/hpcaitech/ColossalAI/wiki/2.-Developer-Guideline

Reply:
> Hey, @SauravMaheshkar , you can join our development channel via the link on this page.
> 
> https://github.com/hpcaitech/ColossalAI/wiki/2.-Developer-Guideline

Joined 👍🏽 @FrankLeeeee

Reply:
Great.

Reply:
I have tried your commit, seem it is not working as expected. When running `docker inspect`, I get the following output.

<img width="998" alt="Screenshot 2023-03-13 at 10 16 15" src="https://user-images.githubusercontent.com/31818963/224593026-86f26e15-64c2-4fa9-93d9-db64db1f53fe.png">

As you can see `org.opencontainers.image.version` does not actually resolve the bash command.

Reply:
@FrankLeeeee tbh this makes sense. I checked my reference and even that codebase dropped this after a while. My suggestion would be to drop the `org.opencontainers.image.version` label for now, and rather version the images using docker tags.

Reply:
Ok, that's fine for me.